monkestation/code/modules/antagonists/bloodling/mobs/bloodling_mob.dm

```diff
@@ -0,0 +1,251 @@
+	SIGNAL_HANDLER
+
+	if(biomass <= 0)
+		gib()
+		return
+
+// Bloodlings health and damage needs updating when biomass is added
+/mob/living/basic/bloodling/proper/add_biomass(amount)
+	. = ..()
+	// Damage is based on biomass, and handled here
+	obj_damage = biomass * 0.2
+	// less than 5 damage would be very bad
+	if(biomass > 50)
+		melee_damage_lower = biomass * 0.1
+		melee_damage_upper = biomass * 0.1
+	update_health_hud()
+	check_evolution()
+
+/// Checks if we should evolve, and also calls the evolution proc
+/mob/living/basic/bloodling/proper/proc/check_evolution()
+	if(75 > biomass && evolution_level != 1)
+		evolution(1)
+		return
+	if(125 > biomass >= 75 && evolution_level != 2)
+		evolution(2)
+		return
+	if(175 > biomass >= 125 && evolution_level != 3)
+		evolution(3)
+		return
+	if(225 > biomass >= 175 && evolution_level != 4)
+		evolution(4)
+		return
+	if(biomass >= 225 && evolution_level != 5)
+		evolution(5)
+		return
+
+/// Creates the mob for us to then mindswap into
+/mob/living/basic/bloodling/proper/proc/evolution(tier)
+	var/new_bloodling = null
+	switch(tier)
+		if(1)
+			new_bloodling = new /mob/living/basic/bloodling/proper/tier1/(src.loc)
+		if(2)
+			new_bloodling = new /mob/living/basic/bloodling/proper/tier2(src.loc)
+		if(3)
+			new_bloodling = new /mob/living/basic/bloodling/proper/tier3(src.loc)
+		if(4)
+			new_bloodling = new /mob/living/basic/bloodling/proper/tier4(src.loc)
+		if(5)
+			new_bloodling = new /mob/living/basic/bloodling/proper/tier5(src.loc)
+	evolution_mind_change(new_bloodling)
+
+
+/mob/living/basic/bloodling/proper/proc/evolution_mind_change(var/mob/living/basic/bloodling/proper/new_bloodling)
+	visible_message(
+		span_alertalien("[src] begins to grow!"),
+		span_noticealien("You evolve!"),
+	)
+	new_bloodling.setDir(dir)
+	if(numba)
+		new_bloodling.numba = numba
+		new_bloodling.set_name()
+	new_bloodling.name = name
+	new_bloodling.real_name = real_name
+	if(mind)
+		mind.name = new_bloodling.real_name
+		mind.transfer_to(new_bloodling)
+	new_bloodling.add_biomass(biomass)
+	qdel(src)
+
+/// Our health hud is based on biomass, since our health is infinite
+/mob/living/basic/bloodling/proper/update_health_hud()
+	if(isnull(hud_used))
+		return
+
+	hud_used.healths.maptext = MAPTEXT("<div align='center' valign='middle' style='position:relative; top:0px; left:6px'><font color='red'>[biomass]E</font></div>")
+
+/// Checks for damage to update the bloodlings biomass accordingly
+/mob/living/basic/bloodling/proper/proc/on_damaged(datum/source, damage, damagetype)
+	SIGNAL_HANDLER
+
+	// Stamina damage is fucky, so we ignore it
+	if(damagetype == STAMINA)
+		return
+
+	// Bloodlings take damage through their biomass, not regular damage
+	add_biomass(-damage)
+
+/mob/living/basic/bloodling/proper/Destroy()
+	UnregisterSignal(src, COMSIG_LIVING_LIFE)
+	UnregisterSignal(src, COMSIG_MOB_APPLY_DAMAGE)
+
+	. = ..()
```

Comment:
quite sure you can replace with `return ..()` since its at tha end

Reply:
Yeah